arithmetization/src/main/java/net/consensys/linea/zktracer/module/limits/precompiles/RipeMd160Blocks.java

```diff
@@ -38,12 +50,9 @@ public String moduleKey() {
     return "PRECOMPILE_RIPEMD_BLOCKS";
   }
 
-  private static final int PRECOMPILE_BASE_GAS_FEE = 600;
-  private static final int PRECOMPILE_GAS_FEE_PER_EWORD = 120;
-  private static final int RIPEMD160_BLOCKSIZE = 64 * 8;
-  // If the length is > 2⁶4, we just use the lower 64 bits.
-  private static final int RIPEMD160_LENGTH_APPEND = 64;
-  private static final int RIPEMD160_ND_PADDED_ONE = 1;
+  @Getter private final ShakiraData data;
+
+  private int lastDataCallHubStamp = 0;
```

Comment:
It's very strange that this info is stored here ... Amha, the last hubStamp should be in ShakiraOperation only, with somethink like `previousHubStamp = shakiraStamp ==1 ? 0 : shakiraOperations.get(shakiraStamp-2).hubStamp()`

Reply:
I concur with @letypequividelespoubelles in the observation that it's the Shakira class that ought to know the previous ID (rather than objects related to either KECCAK, SHA2-256 or RIPEMD-160. It's module wide shared information and bears no particular connection to the nature of the operation.

On a different note I believe we don't care about the `previousHubStamp` as much as we care about the `previousId`. In any case the SHAKIRA module knows only about the ID.